src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -946,7 +946,95 @@ public String Remove(int startIndex, int count)
             Contract.EndContractBlock();
 
             return Substring(0, startIndex);
-        }   
+        }
+
+        public string Replace(string oldValue, string newValue, bool ignoreCase, CultureInfo culture)
```

Comment:
not sure, the checks are done in ReplaceCore. ReplaceCore will throw when oldValue == null